examples/src/sections/CharitableGiving.tsx

```diff
@@ -27,58 +38,106 @@ const TOKENS = [
+    contractAddress: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
+  },
+  {
+    symbol: 'USDC',
+    decimals: 6,
+    contractAddress: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
+  },
+  {
+    symbol: 'DAI',
+    decimals: 18,
+    contractAddress: '0x6b175474e89094c44da98b954eedeac495271d0f',
+  },
 ];
 
 function CharitableGiving({ sdk }: { sdk: EndaomentSdkApi }) {
+  const { address } = useAccount();
+
   const [ein, setEin] = useState('844661797');
   const [selectedToken, setSelectedToken] = useState(TOKENS[0]);
   const [amountIn, setAmountIn] = useState('1');
-  const [exp, setExp] = useState(1e18);
+  const [amountInParsed, setAmountInParsed] = useState(parseUnits(amountIn || '0', selectedToken.decimals));
   const [loading, setLoading] = useState(false);
   const [swapAndDonateTransaction, setSwapAndDonateTransaction] = useState<NdaoSdkDonationSwap>();
-  const { config, error } = usePrepareSendTransaction({
+
+  // Send swap and donate transaction
+  const { config: config1 } = usePrepareSendTransaction({
     request: {
       to: swapAndDonateTransaction?.to as string,
       data: swapAndDonateTransaction?.data as string,
       value: swapAndDonateTransaction?.value as string,
-      gasLimit: 500_000,
+      gasLimit: 1_000_000,
     },
+    enabled: !!swapAndDonateTransaction,
     onError: (error) => console.error(error),
   });
-  const { sendTransaction } = useSendTransaction(config);
+  const { sendTransaction } = useSendTransaction(config1);
+
+  const isETH = selectedToken.symbol === 'ETH';
+
+  const { data: entityAllowance } = useContractRead({
+    address: selectedToken.contractAddress as Address,
+    abi: erc20ABI,
+    functionName: 'allowance',
+    args: [address as Address, swapAndDonateTransaction?.to as Address],
+    enabled: !isETH && !!address && !!swapAndDonateTransaction?.to,
+    watch: true,
+  });
+
+  const { config: config2 } = usePrepareContractWrite({
+    address: selectedToken.contractAddress as Address,
+    abi: erc20ABI,
+    functionName: 'approve',
+    args: [swapAndDonateTransaction?.to as Address, amountInParsed],
+    enabled: !isETH && !!address && !!swapAndDonateTransaction?.to,
+    overrides: {
+      gasLimit: BigNumber.from(100_000),
+    },
+  });
+  const { write: erc20Approve } = useContractWrite(config2);
+
+  const requiresAllowance = entityAllowance && entityAllowance.lt(amountInParsed);
+
+  console.log(amountIn, amountInParsed, entityAllowance, requiresAllowance);
+
+  const handleTokenChange = (v: any) => {
+    setSwapAndDonateTransaction(undefined);
+    setSelectedToken(TOKENS.find((t) => t.contractAddress === v.target.value)!);
+  };
+
+  const handleAmountInChange = (v: string) => {
+    if (isNaN(+v)) return;
+    setSwapAndDonateTransaction(undefined);
+
+    setAmountIn(v);
+    setAmountInParsed(parseUnits(v || '0', selectedToken.decimals));
+  };
+
+  const handleApproveAllowance = () => {
+    if (requiresAllowance && erc20Approve) erc20Approve();
+  };
```

Comment:
Attempting to donate USDC gives me an infinite loader

Reply:
Please see PR description (just updated)